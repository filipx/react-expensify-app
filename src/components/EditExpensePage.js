import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expensesActions';

export class EditExpensePage extends React.Component {
  onEditExpense = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onRemoveExpense = () => {
    this.props.removeExpense({ id : this.props.expense.id });
    this.props.history.push('/');
  }
  render() {
    return (
      <div>
        <ExpenseForm 
          expense={this.props.expense} 
          onSubmit={this.onEditExpense}
        />
        <br />
        <button onClick={this.onRemoveExpense}>Remove Expense</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (id) => dispatch(removeExpense(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);