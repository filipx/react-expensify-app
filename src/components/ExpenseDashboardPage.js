import React from 'react';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseList from './ExpenseList';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;