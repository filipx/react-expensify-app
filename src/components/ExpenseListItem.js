import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <div>
    <h2><Link to={`/edit/${id}`}>{description}</Link></h2>
    <p>{amount / 100}$ {createdAt > 0 && <small>- at: {createdAt}</small>}</p>
  </div>
);

export default ExpenseListItem;