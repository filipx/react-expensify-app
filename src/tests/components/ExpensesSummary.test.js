import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpenseSummary with 1 expense corretly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses corretly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2354544} />);
  expect(wrapper).toMatchSnapshot();
});