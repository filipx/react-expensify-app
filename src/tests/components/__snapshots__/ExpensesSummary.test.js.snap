// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense corretly 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h2>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses corretly 1`] = `
<div>
  <h2>
    Viewing 
    23
     
    expenses
     totalling 
    $23,545.44
  </h2>
</div>
`;
